src/pipelines/floodscan_pipeline.py

```diff
@@ -0,0 +1,437 @@
+                    )
+                except Exception as err:
+                    self.logger.error(f"Failed downloading: {err}")
+            else:
+                zipped_files_path.append(
+                    {SFED: sfed_local_file_path, MFED: mfed_local_file_path}
+                )
+
+        return zipped_files_path
+
+    def _unzip_90days_file(self, file_to_unzip, dates):
+        unzipped_files = []
+        try:
+            with ZipFile(file_to_unzip, "r") as zipobj:
+                for filename in zipobj.namelist():
+                    if os.path.basename(filename):
+                        date = get_datetime_from_filename(filename)
+                        if date in dates:
+                            date_str = re.search("([0-9]{4}[0-9]{2}[0-9]{2})", filename)
+                            new_filename = os.path.basename(
+                                filename.replace(
+                                    date_str[0], date.strftime(DATE_FORMAT)
+                                )
+                            )
+                            try:
+                                full_path = zipobj.extract(filename, self.local_raw_dir)
+                                new_full_path = os.path.join(
+                                    os.path.dirname(full_path), new_filename
+                                )
+                                os.rename(full_path, new_full_path)
+                                unzipped_files.append(
+                                    os.path.basename(
+                                        shutil.move(new_full_path, self.local_raw_dir)
+                                    )
+                                )
+                            except FileExistsError:
+                                self.logger.warning(
+                                    f"File already exists: {new_filename}"
+                                )
+                            except Exception as e:
+                                self.logger.error(f"Failed to extract: {e}")
+
+            return unzipped_files
+        except Exception as e:
+            self.logger.error(f"Failed to extract: {e}")
+
+    def process_historical_data(self, filepath, dates, band_type):
+        paths = {}
+
+        self.logger.info(f"Processing historical data from {filepath}")
+        with xr.open_dataset(filepath) as ds:
+            ds = ds.transpose("time", "lat", "lon")
+            if not ds["time"].dtype == "<M8[ns]":
+                ds["time"] = pd.to_datetime(
+                    [pd.Timestamp(t.strftime(DATE_FORMAT)) for t in ds["time"].values]
+                )
+            for date in dates:
+                if date.year < 2024:
+                    ds_sel = ds.sel({"time": date})
+                    ds_sel = ds_sel.rename({band_type + "_AREA": band_type})
+                    da = ds_sel[band_type]
+                    self.metadata["units"] = "Flood Fraction"
```

Comment:
You should be able to remove all hard-coded `metadata` field assignments here (and elsewhere in the code). These will be set automatically based on the entries in the `floodscan_config.yml` file, since you have the assignment [here](https://github.com/OCHA-DAP/ds-raster-pipelines/blob/6c8f9406b4b5f119e355dba4bca58a42b39533e0/src/pipelines/floodscan_pipeline.py#L36).